cryptoki/tests/basic.rs

```diff
@@ -687,6 +687,91 @@ fn derive_key() -> TestResult {
     Ok(())
 }
 
+pub fn dehex(s: &str) -> Result<Vec<u8>, std::num::ParseIntError> {
+    assert_eq!(s.len() % 2, 0);
+
+    let l = s.len() / 2;
+    let mut result = Vec::with_capacity(l);
+    for i in 0..l {
+        result.push(u8::from_str_radix(&s[i * 2..i * 2 + 2], 16)?);
+    }
+    Ok(result)
+}
+
+#[test]
+#[serial]
+fn derive_key_sp800() -> TestResult {
+    if is_softhsm() {
+        return Ok(());
+    }
+
+    use cryptoki::mechanism::elliptic_curve::*;
+
+    let (pkcs11, slot) = init_pins();
+
+    // open a session
+    let session = pkcs11.open_rw_session(slot)?;
+
+    // log in the session
+    session.login(UserType::User, Some(&AuthPin::new(USER_PIN.into())))?;
+
+    // sha256_sp800
+    let key = dehex("F00670C5F139E7E6C2511EA04FF507AFBBE237CE3C71A89CA59A1C5CF8856562").expect("valid hex");
+    let ephemeral_key = dehex("533B3F09E53B3DEED661A13F7A7D9694AB71CE156C728E00DEE87A1EE3A14C4A").expect("valid hex");
+    let kdf_param = dehex("0A2B0601040197550105011203010807416E6F6E796D6F75732053656E646572202020205633A4C5AE4305BC0FE2ABB699A8EE54632790A0").expect("valid hex");
+    let derivation = dehex("AF8CE51D0139A6D60831A9BABAB20186").expect("valid hex");
```

Comment:
I would use `hex::decode()` as we have `hex` already in the list of dependencies (but looks unused though).